graphein/datasets/pdb_data.py

```diff
@@ -0,0 +1,1092 @@
+            )
+            # Sequence-based ratio splits
+            if split_ratios is not None:
+                assert len(splits) == len(split_ratios)
+                assert sum(split_ratios) == 1.0
+                self.split_ratios = split_ratios
+            # Time-based splits
+            if split_time_frames is not None:
+                frames_are_sequential = True
+                last_frame_index = len(split_time_frames) - 1
+                for frame_index in range(len(split_time_frames)):
+                    frame = split_time_frames[frame_index]
+                    frames_are_backwards_sequential = frame_index == 0 or (
+                        frame_index > 0
+                        and frame > split_time_frames[frame_index - 1]
+                    )
+                    frames_are_forwards_sequential = (
+                        frame_index < last_frame_index
+                        and frame < split_time_frames[frame_index + 1]
+                    ) or frame_index == last_frame_index
+                    frames_are_sequential = all(
+                        [
+                            frames_are_backwards_sequential,
+                            frames_are_forwards_sequential,
+                        ]
+                    )
+                assert len(splits) == len(split_time_frames)
+                assert frames_are_sequential
+                self.split_time_frames = split_time_frames
+
+    def download_metadata(self):
+        """Download all PDB metadata."""
+        self._download_pdb_sequences()
+        self._download_ligand_map()
+        self._download_source_map()
+        self._download_resolution()
+        self._download_entry_metadata()
+
+    @property
+    def num_unique_pdbs(self) -> int:
+        """Return the number of unique PDB IDs in the dataset.
+
+        :return: Number of unique PDB IDs.
+        :rtype: int
+        """
+        return len(self.df.pdb.unique())
+
+    @property
+    def unique_pdbs(self) -> List[str]:
+        """Return a list of unique PDB IDs in the dataset.
+
+        :return: List of unique PDB IDs.
+        :rtype: List[str]
+        """
+        return self.df.pdb.unique().tolist()
+
+    @property
+    def num_chains(self) -> int:
+        """Return the number of chains in the dataset.
+
+        :return: Number of chains.
+        :rtype: int
+        """
+        return len(self.df)
+
+    @property
+    def longest_chain(self) -> int:
+        """Return the length of the longest chain in the dataset.
+
+        :return: Length of the longest chain.
+        :rtype: int
+        """
+        return self.df.length.max()
+
+    @property
+    def shortest_chain(self) -> int:
+        """Return the length of the shortest chain in the dataset.
+
+        :return: Length of the shortest chain.
+        :rtype: int
+        """
+        return self.df.length.min()
+
+    @property
+    def highest_resolution(self) -> float:
```

Comment:
What do you think about renaming "highest" and "lowest" to "best" and "worst" - it could possibly be a little confusing otherwise?

Reply:
Great point. I've improved the verbiage surrounding PDB resolutions in [this commit](https://github.com/a-r-j/graphein/pull/272/commits/67bc7886fe5d9141faf639295b92356154d259b7).